index.js

```diff
@@ -21,33 +18,25 @@ fs.mkdir(pathName, { recursive: true }, (error) => {
 
 fs.readFile(file, 'utf8', (err, data) => {
     if (err) {
-        return console.log(err)
+        throw err;
     }
 
     // the data is split into a new array called clippings
     const clippings = data.split('==========');
 
     // declare an empty array that will hold the books & highlights
-    let books = [];
-
-    // searchBooks searches in 'books' array & returns index of book if it exists
-    function searchBooks(books, title) {
-        for(let i = 0; i < books.length; i++) {
-            if (books[i].title === title) {
-                bookIndex = i;
-                return;
-            }
-        }
-        bookIndex = -1;
-        return bookIndex;
-    }
+    const books = {};
 
     for(let i = 0; i < clippings.length; i++) {
 
         // split each clipping in clippings into component parts whenever there's a new line
-        let clipping = clippings[i].split('\r\n');
+        let clipping = clippings[i].split('\n');
         clipping = clipping.filter(e => e.length);
 
+        // in case the entry is blank (seems like the last entry is always blank based on the txt file format)
+        if (clipping.length === 0) {
+          continue; // a way to skip a single iteration of a loop;
+        }
```

Comment:
this is for situations in which filtering all the lines of a clipping results in an array with nothing in it.